Fixed help command

AndrewJWin opened the pull request:
They are as such:

-  Fixed filtering for certain categories, main one being `social`.
-  Added Owner only override for NSFW command listing.
-  Fixed miscellaneous issues that occurred while doing prior fixes.

AndrewJWin committed:
Fixed filter and included Owner override

AndrewJWin committed:
Fixes filtering and other issues

Fixed a couple issues, also set up filtering for NSFW commands in a NSFW channel
* Overridden for Owner
For categories; commands that the user cannot use, but if there are commands available, the remaining commands can still be viewed.

MrJacz commented on commands/general/help.js:
```diff
@@ -32,50 +33,47 @@ class Help extends Command {
-          n++;
-        }
-      });
+      if (this.client.commands.has(type) || this.client.commands.some(command => command.conf.aliases.includes(type))) {
```
you do know that `this.client.aliases` is a thing?

MrJacz commented on commands/general/help.js:
```diff
@@ -32,50 +33,47 @@ class Help extends Command {
+        embed.setTitle(cm.help.name.toProperCase())
+          .addField("Command description", cm.help.description)
+          .addField("Command usage", `\`${cm.help.usage}\``)
+          .addField("Command aliases", cm.conf.aliases.length == 0 ? "None" : cm.conf.aliases.join(", ") );
```
use `===`

MrJacz commented on commands/general/help.js:
```diff
@@ -32,50 +33,47 @@ class Help extends Command {
+      } else {
+        let n = 0;
+        sorted.forEach(c => {
+          if (c.help.category.toLowerCase() === type) {
```
this can be a one liner

MrJacz commented on commands/general/help.js:
```diff
@@ -32,50 +33,47 @@ class Help extends Command {
-            output += `\n\`${message.settings.prefix + c.help.name}\` | ${c.help.description.length > 50 ? c.help.description.slice(0,50) +"...": c.help.description}`;
+        let output = "";
+        let num = 0;
+        const pg = parseInt(page) && parseInt(page) <= Math.ceil(n / perpage) ? parseInt(page) : 1;
```
parse the number once instead of doing it multiple time

AndrewJWin committed:
Fixes

AndrewJWin commented:
-  Fixed L40-C69 to check against aliases.
-  Fixed L46-C61 to strict match case.
-  Fixed L50-L51 to consolidate into one line.
-  Fixed multiple parseInt(page) to a single parseInt() at the beginning of the code.

AndrewJWin committed:
Missed an = while counting characters.

AndrewJWin committed:
New command; Vaportype

recreates the meme to:

**S P A C E  E V E R Y T H I N G  O U T**

AndrewJWin commented:
Included is the a new command: **Vaportype**.

-  Spaces out a given word/sentence.
-  Automagically changes to uppercase.
-  Looks super cool.

YorkAARGH commented:
Can you make the vaportype a separate PR please?

MrJacz commented on commands/fun/vaportype.js:
```diff
@@ -0,0 +1,24 @@
+  }
+
+  async run(message, args, level) { // eslint-disable-line no-unused-vars
+    args[0] ? this.space(message, args) : message.reply("you have to provide me text to space out you baka.");
```
You should be checking args then returning that message instead

AndrewJWin replied:
I am, with ``args[0] ? this.space(message, args) : message.reply("you have to provide me text to space out you baka.");`` it verifies ``args[0]`` with any value provided, which means if you don't have anything, i.e. ``m.vaportype``, it'll respond with the ``"you have to provide me text to space out you baka."``.

MrJacz commented on commands/fun/vaportype.js:
```diff
@@ -0,0 +1,24 @@
+
+  async space(message, args) {
+    let msg = "";
+    await args.forEach(c => msg += c.toUpperCase().split("").join(" ") + "  ");
```
forEach returns void which makes it un-`await`able

MrJacz commented on commands/fun/vaportype.js:
```diff
@@ -0,0 +1,24 @@
+  async space(message, args) {
+    let msg = "";
+    await args.forEach(c => msg += c.toUpperCase().split("").join(" ") + "  ");
+    message.channel.send(msg);
```
return

MrJacz commented on commands/fun/vaportype.js:
```diff
@@ -0,0 +1,24 @@
+    args[0] ? this.space(message, args) : message.reply("you have to provide me text to space out you baka.");
+  }
+
+  async space(message, args) {
```
Also, this function really doesn't need to be a thing, considering few things it does.

AndrewJWin replied:
I wanted to get practice with class functions.

MrJacz commented on commands/fun/vaportype.js:
```diff
@@ -0,0 +1,24 @@
+    });
+  }
+
+  async run(message, args, level) { // eslint-disable-line no-unused-vars
```
this can be `async run(message, args) {`

AndrewJWin replied:
Wasn't sure if that would work with how we execute commands, went ahead and did it anyway.

MrJacz reviewed (changes requested):
The simple use of a for loop would be the suggested way to add the spaces...

MrJacz commented on commands/general/help.js:
```diff
@@ -16,13 +16,17 @@ class Help extends Command {
 
   async run(message, [type, page], level) {
 
+    type ? type = type.toLowerCase() : undefined;
```
Assigning with a ternary is a bad practice so `type = type ? type.toLowerCase() : null`
Use `null` instead (More style than anything tho)

AndrewJWin replied:
No, because it needs to be ``undefined``, not ``null``.

kyranet replied:
@MrJacz is right.

```javascript
if (type) type = type.toLowerCase();
```

MrJacz commented on commands/general/help.js:
```diff
@@ -16,13 +16,17 @@ class Help extends Command {
   async run(message, [type, page], level) {
 
+    type ? type = type.toLowerCase() : undefined;
+    page ? page = parseInt(page) : undefined;
```
Same here

AndrewJWin replied:
Same with above.

MrJacz commented on commands/general/help.js:
```diff
@@ -32,52 +36,47 @@ class Help extends Command {
-          n++;
-        }
-      });
+      if (this.client.commands.has(type) || this.client.aliases.has(type)) {
```
A simple if no then return check would be better in this case

AndrewJWin replied:
This is just fixing the currently uploaded version, no need for major changes.

MrJacz replied:
This wouldnt be major.

AndrewJWin committed:
Fixes

AndrewJWin committed:
Fixes

AndrewJWin committed:
🤦

MrJacz commented on commands/fun/vaportype.js:
```diff
@@ -0,0 +1,24 @@
+
+  async space(message, args) {
+    let msg = "";
+    for (const c of args) { msg += c.toUpperCase().split("").join(" ") + "  "; }
```
The brackets are useless

AndrewJWin committed:
Update vaportype.js

AndrewJWin committed:
Jacz nanoseconds

AndrewJWin committed:
Update vaportype.js
